artifact/gem5art/artifact/_artifactdb.py

```diff
@@ -179,17 +179,17 @@ def searchByLikeNameType(self, name: str, typ: str, limit: int) -> Iterable[Dict
         for d in data:
             yield d
 
-_db = None
+db = None
 
-def getDBConnection() -> ArtifactDB:
+def getDBConnection(typ = ArtifactMongoDB) -> ArtifactDB:
```

Comment:
This needs a type annotation. I think you should do this by following: https://www.python.org/dev/peps/pep-0484/#the-type-of-class-objects

Reply:
```
DB = TypeVar('DB', bound=ArtifactDB)
def getDBConnection(typ = ArtifactMongoDB: Type[DB]) -> DB:
```

Reply:
Try this:
```
def getDBConnection(typ = ArtifactMongoDB: Type[ArtifactDB]) -> ArtifactDB:
```

Reply:
Fixed.